libc-bottom-half/headers/public/wasi/preview2.h

```diff
@@ -0,0 +1,2481 @@
+extern bool streams_method_input_stream_read(streams_borrow_input_stream_t self, uint64_t len, streams_list_u8_t *ret, streams_stream_error_t *err);
+// Read bytes from a stream, after blocking until at least one byte can
+// be read. Except for blocking, behavior is identical to `read`.
+extern bool streams_method_input_stream_blocking_read(streams_borrow_input_stream_t self, uint64_t len, streams_list_u8_t *ret, streams_stream_error_t *err);
+// Skip bytes from a stream. Returns number of bytes skipped.
+// 
+// Behaves identical to `read`, except instead of returning a list
+// of bytes, returns the number of bytes consumed from the stream.
+extern bool streams_method_input_stream_skip(streams_borrow_input_stream_t self, uint64_t len, uint64_t *ret, streams_stream_error_t *err);
+// Skip bytes from a stream, after blocking until at least one byte
+// can be skipped. Except for blocking behavior, identical to `skip`.
+extern bool streams_method_input_stream_blocking_skip(streams_borrow_input_stream_t self, uint64_t len, uint64_t *ret, streams_stream_error_t *err);
+// Create a `pollable` which will resolve once either the specified stream
+// has bytes available to read or the other end of the stream has been
+// closed.
+// The created `pollable` is a child resource of the `input-stream`.
+// Implementations may trap if the `input-stream` is dropped before
+// all derived `pollable`s created with this function are dropped.
+extern streams_own_pollable_t streams_method_input_stream_subscribe(streams_borrow_input_stream_t self);
+// Check readiness for writing. This function never blocks.
+// 
+// Returns the number of bytes permitted for the next call to `write`,
+// or an error. Calling `write` with more bytes than this function has
+// permitted will trap.
+// 
+// When this function returns 0 bytes, the `subscribe` pollable will
+// become ready when this function will report at least 1 byte, or an
+// error.
+extern bool streams_method_output_stream_check_write(streams_borrow_output_stream_t self, uint64_t *ret, streams_stream_error_t *err);
+// Perform a write. This function never blocks.
+// 
+// Precondition: check-write gave permit of Ok(n) and contents has a
+// length of less than or equal to n. Otherwise, this function will trap.
+// 
+// returns Err(closed) without writing if the stream has closed since
+// the last call to check-write provided a permit.
+extern bool streams_method_output_stream_write(streams_borrow_output_stream_t self, streams_list_u8_t *contents, streams_stream_error_t *err);
+// Perform a write of up to 4096 bytes, and then flush the stream. Block
+// until all of these operations are complete, or an error occurs.
+// 
+// This is a convenience wrapper around the use of `check-write`,
+// `subscribe`, `write`, and `flush`, and is implemented with the
+// following pseudo-code:
+// 
+// ```text
+// let pollable = this.subscribe();
+// while !contents.is_empty() {
+  // // Wait for the stream to become writable
```

Comment:
The indentation looks like it gets confused here..

Reply:
Yeah, `wit-bindgen` has a very primitive indentation algorithm which is easily fooled, unfortunately.   Perhaps running the output through `clang-format` would be enough to fix the damage.

Reply:
https://github.com/bytecodealliance/wit-bindgen/pull/804